src/client/minigames/PasswordGame/PasswordGame.tsx

```diff
@@ -0,0 +1,102 @@
+import * as Owasp from 'owasp-password-strength-test'
+import * as React from 'react'
+import { Alert, Form, FormGroup, Input, Label, Progress } from 'reactstrap'
+
+const styles = require('./scss/PasswordGame.scss') // tslint:disable-line no-var-requires
+
+interface State {
+  first: string,
+  passed: number,
+  percent: number,
+}
+
+class PasswordGame extends React.Component<{}, State> {
+
+  constructor(props: {} = {}) {
+    super(props)
+    this.state = {
+      first : '',
+      passed: 0,
+      percent: 0,
+    }
+  }
+
+  componentDidMount = () => {
+    Owasp.
+      config({
+        allowPassphrases: true,
+        maxLength: 128,
+        minLength: 10,
+        minPhraseLength: 20,
+        minOptionalTestsToPass: 4,
+      })
+  }
+
+  firstChange = (e: React.SyntheticEvent) => {
+    const first = (e.target as HTMLInputElement).value
+    console.log(first)
+    const passed = Owasp.test(first).passedTests.length
+    const failed = Owasp.test(first).failedTests.length
+    const total = passed + failed
+
+    const percent = Owasp.test(first).passedTests.includes(0)
+      ? passed / total * 100
+      : first.length
+    this.setState({
```

Comment:
I know this is nitpicky, but I would remove the line before `const percent...` and add one before the `this.setState...`. The reason being that we separate the variable declarations and the set state call.